### PR TITLE
[Numpy] fix test_operator_gpu.test_upsampling_bilinear_with_type

### PR DESCRIPTION
**Pull request by junrushao1994**
Hey @reminisce yet another out-of-bound error

### CODE REVIEW

File: src/operator/nn/deconvolution-inl.h [hunk 1]
```diff
@@ -122,7 +122,7 @@ struct DeconvolutionParam : public dmlc::Parameter<DeconvolutionParam> {
     // Modified by Li.bs
     // Use tag to control the calculation of pad
     bool bCal = false;
-    if (target_shape.ndim() != 0) {
```
Comment (reminisce) [inline src/operator/nn/deconvolution-inl.h]: `target_shape` is default to `mxnet::TShape(0)`, so the original code is correct.

File: src/operator/nn/deconvolution-inl.h [hunk 2]
```diff
@@ -143,9 +143,9 @@ struct DeconvolutionParam : public dmlc::Parameter<DeconvolutionParam> {
         }
       }
     } else {
-      for (size_t i = 0; i < ndim; i++) {
-        o_pad[i] = pad[i];
-        o_adj[i] = adj[i];
+      for (int i = 0; i < (int) ndim; i++) {
+        o_pad[i] = i < pad.ndim() ? pad[i] : 0;
```
Comment (reminisce) [inline src/operator/nn/deconvolution-inl.h]: According the [initialization function](https://github.com/apache/incubator-mxnet/blob/master/src/operator/nn/deconvolution.cc#L335), `pad.ndim()` should be equal to `ndim` here. Do you understand why `i` can be bigger than `pad.ndim()`?

**Comment by abhinavs95** [comment]
@mxnet-label-bot add [numpy]

File: src/operator/nn/deconvolution-inl.h (hunk 1 above)
Comment (junrushao1994) [reply src/operator/nn/deconvolution-inl.h]: Correct. Revert this accordingly

File: src/operator/nn/deconvolution-inl.h (hunk 2 above)
Comment (junrushao1994) [reply src/operator/nn/deconvolution-inl.h]: The upsampling operator customized DeconvolutionParam [here](https://github.com/apache/incubator-mxnet/blob/master/src/operator/nn/upsampling-inl.h#L170-L188). It is kinda weird but I believe adj is not used in this case.

**Comment by junrushao1994** [comment]
@reminisce Hey could you review again? Thanks!

**Pull request merged**